dot/rpc/modules/author.go

```diff
@@ -169,17 +169,23 @@ func (cm *AuthorModule) SubmitExtrinsic(r *http.Request, req *Extrinsic, res *Ex
 	vtx := transaction.NewValidTransaction(ext, txv)
 
 	if cm.coreAPI.IsBabeAuthority() {
-		hash, err := cm.txQueueAPI.Push(vtx)
-		if err != nil {
-			log.Trace("[rpc] submitted extrinsic failed to push transaction to queue", "error", err)
+		hash, errQueue := cm.txQueueAPI.Push(vtx)
+		if errQueue != nil {
+			log.Trace("[rpc] submitted extrinsic failed to push transaction to queue", "error", errQueue)
 			return err
```

Comment:
the reason the test fails is here: you are returning err but should return errQueue